src/App.tsx

```diff
@@ -1,14 +1,29 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
-import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { TodoList } from './components/TodoList';
+import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>(todosFromServer);
```

Comment:
You're initializing `filteredTodos` with `todosFromServer` but at the time of initialization, `todosFromServer` is an empty array. This can lead to unexpected behavior because `filteredTodos` won't be updated when `todosFromServer` changes. You should initialize `filteredTodos` as an empty array and update it in a `useEffect` hook when `todosFromServer` changes. You can read more about it [here](https://stackoverflow.com/a/50403930).